src/main/java/cms/gov/madie/measure/resources/MeasureController.java

```diff
@@ -271,4 +271,24 @@ private boolean isMeasurementPeriodChanged(Measure measure, Measure persistedMea
         || !Objects.equals(
             persistedMeasure.getMeasurementPeriodEnd(), measure.getMeasurementPeriodEnd());
   }
+
+  @GetMapping("/measures/search/{criteria}")
+  public ResponseEntity<Page<Measure>> findAllByMeasureNameOrEcqmTitle(
+      Principal principal,
+      @RequestParam(required = false, defaultValue = "false", name = "currentUser")
+          boolean filterByCurrentUser,
+      @PathVariable("criteria") String criteria,
+      @RequestParam(required = false, defaultValue = "10", name = "limit") int limit,
+      @RequestParam(required = false, defaultValue = "0", name = "page") int page) {
+
+    final String username = principal.getName();
+    final Pageable pageReq = PageRequest.of(page, limit, Sort.by("lastModifiedAt").descending());
+
+    Page<Measure> measures =
+        filterByCurrentUser
+            ? repository.findAllByMeasureNameOrEcqmTitleForCurrentUser(criteria, pageReq, username)
+            : repository.findAllByMeasureNameOrEcqmTitle(criteria, pageReq);
```

Comment:
This is fine for now, but as we add more search criteria we may want to move to something like the Criteria class or query by Example.

https://docs.spring.io/spring-data/mongodb/docs/current/reference/html/#mongo.query

https://docs.spring.io/spring-data/mongodb/docs/current/reference/html/#query-by-example

Reply:
Ok thanks!